resource-managers/kubernetes/core/src/main/scala/org/apache/spark/scheduler/cluster/k8s/ExecutorPodsLifecycleManager.scala

```diff
@@ -211,7 +219,9 @@ private[spark] class ExecutorPodsLifecycleManager(
           .pods()
           .inNamespace(namespace)
           .withName(updatedPod.getMetadata.getName)
-        if (podToDelete.get() != null) {
+
+        if (podToDelete.get() != null &&
+            podToDelete.get.getMetadata.getDeletionTimestamp == null) {
```

Comment:
This part of the code has undergone multiple changes, but in practice, it still sends requests to the API server. If we consider first checking the removedExecutorsCache variable (even though it has an expiration time) to determine whether it has already been removed, it would greatly reduce the number of requests to the API server.